### PR TITLE
OBAM-335 UI updates to ward pull/adhoc request

### PR DESCRIPTION
**Pull request by kchelstowski**
I spent a long time trying to figure out how to make the red asterisk for header - the same asterisk which is visible for "regular" fields with required: true. It was supposedly made and was used in many components (talking about fields, which have another nestled fields inside (ArrayField)), but it did not really work with current version, so even though it was used in many places, the asterisk didn't work.
With my changes, the red asterisk for headers are now "generic", which means they can be used anywhere (and will now finally work in places, where it was previously declared (e.g. stock-movement/combined-shipments/AddItemsPage.jsx) and many others, so I hope my changes will also improve other wizards.

### CODE REVIEW

File: src/js/components/stock-movement-wizard/request/AddItemsPage.jsx [hunk 1]
```diff
@@ -617,9 +743,9 @@ class AddItemsPage extends Component {
 
           return {
             ...val,
+            quantityOnHand: '',
             disabled: true,
             quantityRequested: qtyRequested >= 0 ? qtyRequested : 0,
-            quantityOnHand: isPullType ? quantityOnHand : undefined,
```
Comment (awalkowiak) [inline src/js/components/stock-movement-wizard/request/AddItemsPage.jsx:622]: Why did you make this change?

File: src/js/components/stock-movement-wizard/request/AddItemsPage.jsx [hunk 2]
```diff
@@ -716,6 +842,14 @@ class AddItemsPage extends Component {
       if (date.diff(dateRequested) > 0) {
         errors.lineItems[key] = { expirationDate: 'react.stockMovement.error.invalidDate.label' };
       }
+
+      if (this.state.isRequestFromWard) {
+        const isNumeric = n => !Number.isNaN(parseInt(n, 10));
```
Comment (awalkowiak) [inline src/js/components/stock-movement-wizard/request/AddItemsPage.jsx:847]: Why the need for this?

File: src/js/components/stock-movement-wizard/request/AddItemsPage.jsx [hunk 3]
```diff
@@ -716,6 +842,14 @@ class AddItemsPage extends Component {
...
+        const isPositive = n => isNumeric(n) && n >= 0;
+        if (!item.quantityOnHand || !isPositive(item.quantityOnHand)) {
```
Comment (awalkowiak) [inline src/js/components/stock-movement-wizard/request/AddItemsPage.jsx:849]: Why you can't simply do `item.quantityOnHand >= 0`?

**Review by awalkowiak — CHANGES_REQUESTED** [review]

File: src/js/components/stock-movement-wizard/request/AddItemsPage.jsx (hunk 1 above)
Comment (kchelstowski) [reply src/js/components/stock-movement-wizard/request/AddItemsPage.jsx:622]: It was said that the qoh shall be blank, would it be blank in any circumstances if I had left it as it was?

File: src/js/components/stock-movement-wizard/request/AddItemsPage.jsx [hunk 4]
```diff
@@ -716,6 +842,12 @@ class AddItemsPage extends Component {
       if (date.diff(dateRequested) > 0) {
         errors.lineItems[key] = { expirationDate: 'react.stockMovement.error.invalidDate.label' };
       }
+
+      if (this.state.isRequestFromWard) {
+        if (!item.quantityOnHand || !(item.quantityOnHand >= 0)) {
```
Comment (awalkowiak) [inline src/js/components/stock-movement-wizard/request/AddItemsPage.jsx:847]: I think `item.quantityOnHand < 0` would be less confusing

**Review by awalkowiak — CHANGES_REQUESTED** [review]

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**